src/opera/template/relationship.py

```diff
@@ -41,7 +48,22 @@ def get_property(self, params):
         # have data type support.
         if prop not in self.properties:
             raise DataError("Template has no '{}' property".format(prop))
+
         return self.properties[prop].eval(self, prop)
 
+    def get_attribute(self, params):
+        host, attr, *rest = params
+
+        if host != "SELF":
+            raise DataError(
+                "Accessing non-local stuff is bad. Fix your service template."
+            )
+        if host == "HOST":
+            raise DataError("HOST is not yet supported in opera.")
```

Comment:
Move this above `host != "SELF"`, otherwise it's unreachable.

Reply:
Done (included in the latest commit).